node/core/backing/src/lib.rs

```diff
@@ -142,34 +137,44 @@ impl TableContextTrait for TableContext {
 	}
 }
 
-impl TableContext {
-	fn local_id(&self) -> Option<ValidatorId> {
-		self.key.as_ref().map(|k| k.public())
+/// A message type that is sent from `CandidateBackingSubsystem` to `CandidateBackingJob`.
+pub enum ToJob {
```

Comment:
To ensure I didn't accidentally break anything, I retained the public declaration of `CandidateBackingSubsystem`: [old](https://github.com/paritytech/polkadot/pull/1376/files#diff-b2e256a6d7ea7397ed916a1ed6d74934L871) [new](https://github.com/paritytech/polkadot/pull/1376/files#diff-b2e256a6d7ea7397ed916a1ed6d74934R728-R729).

Then I just chased back the compiler errors: whenever rustc complained that a private type was leaking through the interface, I made it public. Ultimately, it's because the types are used in `JobTrait`.

Reply:
So specifically, this is because the `CandidateBackingSubsystem` is a type alias of a generic that is instantiated with private types.

I think it is better to keep as many types private within the crate and only expose the one subsystem type. I suggest that it can be done by making `CandidateBackingSubsystem` a struct that wraps the `JobManager` as opposed to a type alias of it.

Reply:
e3f76a44